src/query/src/promql/planner.rs

```diff
@@ -1907,6 +1915,22 @@ impl PromPlanner {
         Ok((exprs, new_tags))
     }
 
+    /// Validate label name according to Prometheus specification.
+    /// Label names must match the regex: [a-zA-Z_][a-zA-Z0-9_]*
+    /// Additionally, label names starting with double underscores are reserved for internal use.
+    fn validate_label_name(label_name: &str) -> Result<()> {
```

Comment:
Do we need to validate all labels? For example:

https://github.com/GreptimeTeam/greptimedb/blob/e80e4a9ed72827a534448eb5cc0cb7a1b17111a0/src/query/src/promql/planner.rs#L2021
https://github.com/GreptimeTeam/greptimedb/blob/e80e4a9ed72827a534448eb5cc0cb7a1b17111a0/src/query/src/promql/planner.rs#L2007

If this should be a general rule, we could add it to the PromQL parser:
https://github.com/GreptimeTeam/promql-parser

Reply:
seems no need we have already have the check for others
<img width="1085" height="557" alt="image" src="https://github.com/user-attachments/assets/1e54a463-5adc-4b17-ab50-385e89aa931c" />

Reply:
Ok. So do we need to check the labels in the `label_join` function? Which adds the new labels too.